test/extensions/config_subscription/grpc/new_grpc_mux_impl_test.cc

```diff
@@ -561,6 +565,182 @@ TEST_P(NewGrpcMuxImplTest, Shutdown) {
   // There won't be any unsubscribe messages for the legacy mux either for the same reason
 }
 
+// Validate that an EDS resource is cached if there's a cache.
+TEST_P(NewGrpcMuxImplTest, CacheEdsResource) {
```

Comment:
Do we have any integration tests?

Reply:
GrpcMux doesn't have integration tests. There will be after the feature is fully implemented in the next PR (see [the entire feature implementation](https://github.com/adisuissa/envoy/commit/f0b7ac8ab85776947a8332efe1c3759e8714b9ed)).